testproject/Assets/Tests/Runtime/RpcUserSerializableTypesTest.cs

```diff
@@ -225,62 +225,98 @@ public IEnumerator ExtensionMethodRpcTest()
             var obj = new MyObject(256);
             var obj2 = new MySharedObjectReferencedById(256);
             var obj3 = new MyObjectPassedWithThisRef(256);
+            var intList = new List<int> { 5, 10, 15, 5, 1 };
+            var strList = new List<string> { "foo", "bar", "baz", "qux" };
```

Comment:
nit: Maybe a test for nested generics (`List<List<string>>`) or multiple type generics (`KeyValuePair<K, V>`)? Shouldn't matter with the new code change so not terribly important - but not sure how comprehensive you're looking for the tests to be.

Reply:
In an ideal world I would like to make the tests much more comprehensive here. I'd also like to support fully generic extension methods. In the current world - keeping in mind that I've been out for a month and don't know how priorities might have shifted during that time - I didn't think the engineering effort to make them super comprehensive would be worth it.